api/handlers.go

```diff
@@ -86,6 +100,27 @@ func (sl *skylink) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
+// blocklistGET returns a list of blocked hashes and associated tags
+func (api *API) blocklistGET(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	blocked, err := api.staticDB.BlockedHashes()
+	if err != nil {
+		skyapi.WriteError(w, skyapi.Error{err.Error()}, http.StatusInternalServerError)
+		return
+	}
+
+	hashes := make([]BlockedHash, len(blocked))
+	for i, bh := range blocked {
+		hashes[i] = BlockedHash{
+			Hash: bh.Hash.Hash,
+			Tags: bh.Tags,
+		}
+	}
```

Comment:
This is the kind of thing I'm not convinced we should be doing. This call will return a significant amount of data which we are going to duplicate in memory. Then we are going to serialize it, which will add another copy (and then some) of memory. Before we manage to return from this method we are going to have all the data three times in memory. If we don't do that copying action we are going to only have it twice.

This is not a blocking comment on its own. It's more of a discussion I would like to have about the way we handle database DTOs and API DTOs.

Reply:
I'll resolve this in favor of the other comment